From 215d4a4dcc0867508808856261455a77ffa991c5 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Wed, 26 Jun 2024 16:47:13 +0200
Subject: [PATCH] crashpad: patch out use of an absl internal api

---
 components/crash/core/app/crashpad.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/components/crash/core/app/crashpad.cc b/components/crash/core/app/crashpad.cc
index 9281b367f6d88..9c809142d2d98 100644
--- a/components/crash/core/app/crashpad.cc
+++ b/components/crash/core/app/crashpad.cc
@@ -29,7 +29,9 @@
 #include "build/chromeos_buildflags.h"
 #include "components/crash/core/app/crash_reporter_client.h"
 #include "components/crash/core/common/crash_key.h"
+#if 0
 #include "third_party/abseil-cpp/absl/base/internal/raw_logging.h"
+#endif
 #include "third_party/crashpad/crashpad/client/annotation.h"
 #include "third_party/crashpad/crashpad/client/annotation_list.h"
 #include "third_party/crashpad/crashpad/client/crash_report_database.h"
@@ -57,6 +59,7 @@ crashpad::StringAnnotation<24>& PlatformStorage() {
 
 namespace {
 
+#if 0
 void AbslAbortHook(const char* file,
                    int line,
                    const char* buf_start,
@@ -68,6 +71,7 @@ void AbslAbortHook(const char* file,
   logging::LogMessageFatal check_failure(file, line, logging::LOGGING_FATAL);
   check_failure.stream() << "Check failed: false. " << prefix_end;
 }
+#endif
 
 base::FilePath* g_database_path;
 
@@ -170,11 +174,13 @@ bool InitializeCrashpadImpl(bool initial_client,
   // the same file and line.
   base::debug::SetDumpWithoutCrashingFunction(DumpWithoutCrashing);
 
+#if 0
   // TODO(pbos): Update this to not rely on a _internal namespace once there's
   // a public API in absl::.
   // Note: If this fails to compile because of an absl roll, this is fair to
   // remove if you file a crbug.com/new and assign it to pbos@.
   absl::raw_log_internal::RegisterAbortHook(&AbslAbortHook);
+#endif
 
 #if BUILDFLAG(IS_APPLE)
   // On Mac, we only want the browser to initialize the database, but not the
