From 0f5224764f3b6fdcb173a870a52183d60e585683 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Thu, 25 Apr 2024 18:21:48 +0200
Subject: [PATCH] iwyu/base: missing cstdint include in named_trigger

---
 base/trace_event/named_trigger.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/trace_event/named_trigger.h b/base/trace_event/named_trigger.h
index 5553c38ebb2..bf42fe35e7f 100644
--- a/base/trace_event/named_trigger.h
+++ b/base/trace_event/named_trigger.h
@@ -5,6 +5,7 @@
 #ifndef BASE_TRACE_EVENT_NAMED_TRIGGER_H_
 #define BASE_TRACE_EVENT_NAMED_TRIGGER_H_
 
+#include <cstdint>
 #include <optional>
 #include <string>
 
