From 0ddfa588a01b1e9935067cecbf154d75fb5b03a8 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 28 Sep 2024 23:52:52 +0200
Subject: [PATCH] backport std::hardware_{con,de}structive_interference_size
 for clang <19

---
 include/new | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/third_party/libc++/src/include/new b/third_party/libc++/src/include/new
index 3252b0bb1..56ff4c835 100644
--- a/third_party/libc++/src/include/new
+++ b/third_party/libc++/src/include/new
@@ -349,13 +349,20 @@ template <class _Tp>
 
 #if _LIBCPP_STD_VER >= 17
 
-#  if defined(__GCC_DESTRUCTIVE_SIZE) && defined(__GCC_CONSTRUCTIVE_SIZE)
+// backported from https://github.com/llvm/llvm-project/pull/89446
+#  if !defined(__GCC_DESTRUCTIVE_SIZE) || !defined(__GCC_CONSTRUCTIVE_SIZE)
+#    if defined(__aarch64__)
+#      define __GCC_DESTRUCTIVE_SIZE 256
+#      define __GCC_CONSTRUCTIVE_SIZE 64
+#    else
+#      define __GCC_DESTRUCTIVE_SIZE 64
+#      define __GCC_CONSTRUCTIVE_SIZE 64
+#    endif
+#  endif // !defined(__GCC_DESTRUCTIVE_SIZE) || !defined(__GCC_CONSTRUCTIVE_SIZE)
 
 inline constexpr size_t hardware_destructive_interference_size  = __GCC_DESTRUCTIVE_SIZE;
 inline constexpr size_t hardware_constructive_interference_size = __GCC_CONSTRUCTIVE_SIZE;
 
-#  endif // defined(__GCC_DESTRUCTIVE_SIZE) && defined(__GCC_CONSTRUCTIVE_SIZE)
-
 #endif // _LIBCPP_STD_VER >= 17
 
 _LIBCPP_END_NAMESPACE_STD
